tutorpod_autoscaling/hooks.py

```diff
@@ -0,0 +1,29 @@
+"""
+These hooks are stored in a separate module. If they were included in plugin.py, then
+the pod-autoscaling hooks would be created in the context of some other plugin that imports
+them.
+"""
+
+from __future__ import annotations
+
+import typing as t
+
+from tutor.core.hooks import Filter
+
+AUTOSCALING_ATTRS_TYPE = t.Dict[
+    t.Literal[
+        "enable_hpa"
+        "memory_request",
+        "cpu_request",
+        "memory_limit",
+        "cpu_limit",
+        "min_replicas",
+        "max_replicas",
+        "avg_cpu",
+        "avg_memory",
+        "enable_vpa",
+    ],
+    t.Union["str", int],
```

Comment:
Nice typing job! Alternatively, you could have a `tuple[str, str|int]` to ensure that the right types are set for every key. Something like:

```
t.Union[
    t.Tuple[t.Literal["memory_limit"], int],
   ...
]]
```